src/Sulu/Component/Content/Mapper/ContentMapper.php

```diff
@@ -1640,13 +1640,13 @@ private function getLocalizedUrlsForPage(Page $page, NodeInterface $node, $websp
         $contentType = $this->contentTypeManager->get($property->getContentTypeName());
 
         foreach ($webspace->getAllLocalizations() as $localization) {
-
             // prepare translation vars
             $locale = $localization->getLocalization();
+            $shadowLocale = $this->getShadowLocale($node, $localization->getLocalization());
             $translatedProperty = new TranslatedProperty($property, $locale, $this->languageNamespace);
 
             // state property
-            $propertyTranslator = $this->createPropertyTranslator($localization);
+            $propertyTranslator = $this->createPropertyTranslator($shadowLocale);
```

Comment:
Yeah, that's a bit confusing. Maybe also rename the method name to something like `getActualLocale`? Or do you have a better idea?